domparsing/DOMParser-parseFromString-xml-internal-subset.html

```diff
@@ -0,0 +1,24 @@
+<!DOCTYPE html>
+<title>Parsing and serialization of doctype internal subset</title>
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+<script>
+test(function () {
+  // https://www.w3.org/TR/xml11/#sec-prolog-dtd has intSubset as part of the
+  // syntax, which is not represented in the DocumentType interface. Check that
+  // publicId and systemId are not affected.
+  var doc = new DOMParser().parseFromString('<!DOCTYPE foo [ <!ELEMENT foo (#PCDATA)> ]><foo/>', 'text/xml');
+  var doctype = doc.doctype;
+  assert_equals(doctype.name, 'foo', 'doctype name');
+  assert_equals(doctype.publicId, '', 'doctype publicId');
+  assert_equals(doctype.systemId, '', 'doctype systemId');
+  // Sanity check: the root element is still parsed correctly.
+  var documentElement = doc.documentElement;
```

Comment:
Why a local variable that's only used once?  You can keep it if you prefer it stylistically (so the next line doesn't get too long, etc.).  I'm just commenting in case it's a mistake.